core/trino-main/src/main/java/io/trino/metadata/ResolvedFunction.java

```diff
@@ -181,6 +181,11 @@ public String toString()
 
     public static class ResolvedFunctionDecoder
     {
+        private static final JsonCodec<ResolvedFunction> SERIALIZE_JSON_CODEC = new JsonCodecFactory().jsonCodec(ResolvedFunction.class);
+        private static final ThreadSafeCompressorDecompressor COMPRESSOR_DECOMPRESSOR = new ThreadSafeCompressorDecompressor(ZstdCompressor::new, ZstdDecompressor::new);
+        private static final NonEvictableCache<QualifiedName, ResolvedFunction> resolvedFunctionsCache = buildNonEvictableCache(CacheBuilder.newBuilder().maximumSize(1024));
+        private static final NonEvictableCache<ResolvedFunction, QualifiedName> qualifiedFunctionsCache = buildNonEvictableCache(CacheBuilder.newBuilder().maximumSize(1024));
```

Comment:
Caching of resolved names, caching of qualifiedName field and compressor reuse should be separate commits.

Reply:
Done.